unit-tests/artifact_test.py

```diff
@@ -0,0 +1,52 @@
+##############################################################################
+# (c) Crown copyright Met Office. All rights reserved.
+# For further details please refer to the file COPYRIGHT
+# which you should have received as part of this distribution
+##############################################################################
+
+from pathlib import Path
+from fab.artifact import Artifact, New, Unknown
+
+
+class TestArtifact:
+    def test_constructor(self):
+        test_path = Path('/test/path')
+        artifact = Artifact(test_path,
+                            Unknown,
+                            New)
+
+        assert artifact.location == test_path
+        assert artifact.state is New
+        assert artifact.filetype is Unknown
+        assert artifact.depends_on == []
+        assert artifact.defines == []
+        assert artifact._hash is None
+
+    def test_hash(self, tmp_path: Path):
+        test_path = Path(tmp_path / 'test.foo')
+        test_path.write_text("Lorem ipsum dolor sit")
+        expected_hash = 1463158782
+        artifact = Artifact(test_path,
+                            Unknown,
+                            New)
+        assert artifact._hash is None
+        assert artifact.hash == expected_hash
+        # Check that it is cached
+        test_path.unlink()
+        assert artifact.hash == expected_hash
```

Comment:
You raise an interesting question concerning how the artefact class should respond to the file it represents being removed from under it. I think the current behaviour is acceptable so no change needed.

Reply:
For what it's worth - I don't think we need to handle that situation; if we can't rely on files staying where we put (or find) them then there's a whole host of other things which could cause problems.  We can worry about this later but similar to the other comment above - I couldn't think of another way of checking that the hash is really cached